.travis.yml

```diff
@@ -26,28 +28,31 @@ matrix:
     - compiler: clang
       env: SANITIZER="UBSAN"
     - env: COMPILER=gcc7
+    - env: FORMATTER=ON
 
 install:
   - git clone --depth=1 --branch=tikv-3.0 https://github.com/pingcap/rocksdb.git
   - if [ "${COMPILER}" == gcc7 ]; then
       CC=gcc-7;
       CXX=g++-7;
-      COVERAGE_OPT="-DCODE_COVERAGE=ON";
+      export COVERAGE_OPT="-DCODE_COVERAGE=ON";
     fi
   - if [ ! -z "${BUILD_TYPE}" ]; then
-      BUILD_OPT="-DCMAKE_BUILD_TYPE=${BUILD_TYPE}";
+      export BUILD_OPT="-DCMAKE_BUILD_TYPE=${BUILD_TYPE}";
     else
-      BUILD_OPT="-DCMAKE_BUILD_TYPE=Debug";
+      export BUILD_OPT="-DCMAKE_BUILD_TYPE=Debug";
     fi
   - if [ ! -z "${SANITIZER}" ]; then
-      SANITIZER_OPT="-DWITH_${SANITIZER}=ON";
-      TOOLS_OPT="-DWITH_TITAN_TOOLS=OFF";
+      export SANITIZER_OPT="-DWITH_${SANITIZER}=ON";
+      export TOOLS_OPT="-DWITH_TITAN_TOOLS=OFF";
     fi
 
 script:
-  - cmake . -L -DROCKSDB_DIR=./rocksdb -DTRAVIS=ON ${BUILD_OPT} ${SANITIZER_OPT} ${TOOLS_OPT} ${COVERAGE_OPT}
-  - make -j4
-  - ctest -R titan
+  - if [ -z "${FORMATTER}" ]; then
```

Comment:
should we make format check before all tests? Once the format check is failed, running other tests is a waste of CI compute source.

Reply:
All the jobs will run in parallel until finish anyway. and IMO having tests pass is more important than formatter?